src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -50,11 +51,12 @@ public sealed partial class CSharpCompilation : Compilation
         // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 
         private readonly CSharpCompilationOptions _options;
-        private readonly Lazy<UsingsFromOptionsAndDiagnostics> _usingsFromOptions;
-        private readonly Lazy<ImmutableArray<NamespaceOrTypeAndUsingDirective>> _globalImports;
-        private readonly Lazy<Imports> _previousSubmissionImports;
-        private readonly Lazy<AliasSymbol> _globalNamespaceAlias;  // alias symbol used to resolve "global::".
-        private readonly Lazy<ImplicitNamedTypeSymbol?> _scriptClass;
+        private UsingsFromOptionsAndDiagnostics? _usingsFromOptions;
+        private ImmutableArray<NamespaceOrTypeAndUsingDirective> _globalImports;
+        private Imports? _previousSubmissionImports;
+        private AliasSymbol? _globalNamespaceAlias;  // alias symbol used to resolve "global::".
+
+        private StrongBox<NamedTypeSymbol?>? _scriptClass;
```

Comment:
>StrongBox<NamedTypeSymbol?>? [](http://example.com/codeflow?start=16&length=28)

This looks unnecessary complicated. I think we should use ```NamedTypeSymbol?``` as the type for the field. When `null` is one of the valid calculated results, we usually initialize the field to a sentinel value, which then gets replaced with calculated result. In this case, for example, it looks like we can use ```ErrorTypeSymbol.UnknownResultType``` as the sentinel value.

Reply:
Switched to use this suggestion.